### PR TITLE
Fix Typos and Polish in aws.md

### PR DESCRIPTION
**Pull request by skytin1004**
Fixed grammatical errors in aws.md and polish some existing sentences.

### CODE REVIEW

File: docs/aws.md [hunk 1]
```diff
@@ -364,13 +364,13 @@ For more details, please read [S3 ACL Documentation](https://docs.aws.amazon.com
...
 
-For more information on how S3 scales API QPS, checkout the 2018 re:Invent session on [Best Practices for Amazon S3 and Amazon S3 Glacier]( https://youtu.be/rHeTn9pHNKo?t=3219). At [53:39](https://youtu.be/rHeTn9pHNKo?t=3219) it covers how S3 scales/partitions & at [54:50](https://youtu.be/rHeTn9pHNKo?t=3290) it discusses the 30-60 minute wait time before new partitions are created.
+For more information on how S3 scales API QPS, check out the 2018 re: Invent session on [Best Practices for Amazon S3 and Amazon S3 Glacier]( https://youtu.be/rHeTn9pHNKo?t=3219). At [53:39](https://youtu.be/rHeTn9pHNKo?t=3219) it covers how S3 scales/partitions & at [54:50](https://youtu.be/rHeTn9pHNKo?t=3290) it discusses the 30-60 minute wait time before new partitions are created.
```
[the remark this answers is not in the document]
Comment (skytin1004) [reply docs/aws.md:373]: You're right, I'll fix it again.

File: docs/aws.md [hunk 2]
```diff
@@ -288,25 +288,25 @@ This design has the following benefits:
...
+Iceberg also supports the JDBC catalog which uses a table in a relational database to manage Iceberg tables.
+You can configure to use the JDBC catalog with relational database services like [AWS RDS](https://aws.amazon.com/rds).
 Read [the JDBC integration page](../jdbc/#jdbc-catalog) for guides and examples about using the JDBC catalog.
-Read [this AWS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.Connecting.Java.html) for more details about configuring JDBC catalog with IAM authentication. 
+Read [this AWS documentation](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.Connecting.Java.html) for more details about configuring the JDBC catalog with IAM authentication. 
 
 ### Which catalog to choose?
 
-With all the available options, we offer the following guidance when choosing the right catalog to use for your application:
+With all the available options, we offer the following guidlines when choosing the right catalog to use for your application:
```
[the remark this answers is not in the document]
Comment (nastra) [reply docs/aws.md:298]: what I mean is that the word should be `guidelines` instead of `guidlines`

[the remark this answers is not in the document]
Comment (skytin1004) [reply docs/aws.md:298]: Thank you for your attention to detail. You're right. I'll fix it right away.

File: docs/aws.md [hunk 3]
```diff
@@ -555,7 +555,7 @@ This client factory has the following configurable catalog properties:
 | client.assume-role.arn            | null, requires user input                | ARN of the role to assume, e.g. arn:aws:iam::123456789:role/myRoleToAssume  |
 | client.assume-role.region         | null, requires user input                | All AWS clients except the STS client will use the given region instead of the default region chain  |
 | client.assume-role.external-id    | null                                     | An optional [external ID](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html)  |
-| client.assume-role.timeout-sec    | 1 hour                                   | Timeout of each assume role session. At the end of the timeout, a new set of role session credentials will be fetched through a STS client.  |
+| client.assume-role.timeout-sec    | 1 hour                                   | Timeout of each assume role session. At the end of the timeout, a new set of role session credentials will be fetched through an STS client.  |
```
Comment (amogh-jahagirdar) [inline docs/aws.md:558]: I think we can leave this unchanged. "a STS client" is correct

File: docs/aws.md [hunk 4]
```diff
@@ -611,10 +611,10 @@ Apache HTTP Client has the following configurable properties:
...
-| http-client.apache.use-idle-connection-reaper-enabled | null, enabled by default  | An optional `true/false` setting that decide whether to [use idle connection reaper](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#useIdleConnectionReaper(java.lang.Boolean)) |
+| http-client.apache.tcp-keep-alive-enabled             | null, disabled by default | An optional `true/false` setting that decides whether to enable [tcp keep alive](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#tcpKeepAlive(java.lang.Boolean))                 |
+| http-client.apache.use-idle-connection-reaper-enabled | null, enabled by default  | An optional `true/false` setting that decides whether to [use idle connection reaper](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#useIdleConnectionReaper(java.lang.Boolean)) |
```
Comment (amogh-jahagirdar) [inline docs/aws.md:617]: I think we can remove the "decides", these settings aren't sentient :) . 

How about `An optional `true/false` setting that controls whether [use idle connection reaper] is used`

File: docs/aws.md [hunk 5]
```diff
@@ -198,8 +198,8 @@ With optimistic locking, each table has a version id.
 If users retrieve the table metadata, Iceberg records the version id of that table. 
 Users can update the table as long as the version ID on the server side remains unchanged. 
 If there is a version mismatch, it means that someone else has modified the table before you did. 
-The update attempt fails, because you have a stale version of the table. 
-If this happens, Iceberg refreshes the metadata and checks if there might be potential conflict. 
+The update attempt fails because you have a stale version of the table. 
+If this happens, Iceberg refreshes the metadata and checks if there might be a potential conflict.
```
Comment (amogh-jahagirdar) [inline docs/aws.md:202]: We can further shorten this to `checks if there is a conflict`

**Review by amogh-jahagirdar — COMMENTED** [review]
Thanks for polishing these docs, just some minor comments and then we should be good to merge this imo. CC @jackye1995 if he has any further comments

File: docs/aws.md (hunk 3 above)
Comment (skytin1004) [reply docs/aws.md:558]: Before abbreviations like SEO or UN, write a or an, depending on the sound of the first letter. For example, S is a consonant but pronounced /e/, so you would write "an SEO expert", and U is a vowel but pronounced /j/, so you would write "a UN declaration".

Likewise STS would be pronounced as "ess-tee-ess". In this case S is a consonant but pronounced /e/.


I've included links below that you might find helpful.
https://editorsmanual.com/articles/indefinite-article-a-an-with-abbreviations/

Thank you for your attention to detail.

File: docs/aws.md (hunk 4 above)
Comment (skytin1004) [reply docs/aws.md:617]: Thanks for the advice. I will revise it again based on your advice.

File: docs/aws.md (hunk 5 above)
Comment (skytin1004) [reply docs/aws.md]: I've revised the sentence to be more concise based on your advice, thank you.

**Review by amogh-jahagirdar — APPROVED** [review]
Looks good to me, I'll wait for @nastra if he has any further feedback before merging. Thanks @skytin1004 for the contribution!

File: docs/aws.md [hunk 6]
```diff
@@ -611,10 +610,13 @@ Apache HTTP Client has the following configurable properties:
...
 | http-client.apache.connection-max-idle-time-ms        | null                      | An optional [connection max idle timeout](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#connectionMaxIdleTime(java.time.Duration)) in milliseconds                             |
 | http-client.apache.connection-time-to-live-ms         | null                      | An optional [connection time to live](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#connectionTimeToLive(java.time.Duration)) in milliseconds                                  |
-| http-client.apache.expect-continue-enabled            | null, disabled by default | An optional `true/false` setting that decide whether to enable [expect continue](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#expectContinueEnabled(java.lang.Boolean))       |
+| http-client.apache.expect-continue-enabled            | null, disabled by default | An optional `true/false` setting that controls whether [expect continue] is enabled
```
Comment (nastra) [inline docs/aws.md:613]:
```suggestion
| http-client.apache.expect-continue-enabled            | null, disabled by default | An optional `true/false` setting that controls whether [expect continue](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#expectContinueEnabled(java.lang.Boolean)) is enabled
```

**Review by nastra — CHANGES_REQUESTED** [review]
a few links are broken unfortunately

File: docs/aws.md [hunk 7]
```diff
@@ -611,10 +610,13 @@ Apache HTTP Client has the following configurable properties:
...
-| http-client.apache.tcp-keep-alive-enabled             | null, disabled by default | An optional `true/false` setting that decide whether to enable [tcp keep alive](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#tcpKeepAlive(java.lang.Boolean))                 |
-| http-client.apache.use-idle-connection-reaper-enabled | null, enabled by default  | An optional `true/false` setting that decide whether to [use idle connection reaper](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#useIdleConnectionReaper(java.lang.Boolean)) |
+| http-client.apache.tcp-keep-alive-enabled             | null, disabled by default | An optional `true/false` setting that controls whether [tcp keep alive] is enabled
+(https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/apache/ApacheHttpClient.Builder.html#tcpKeepAlive(java.lang.Boolean))                 |
+| http-client.apache.use-idle-connection-reaper-enabled | null, enabled by default  | An optional `true/false` setting that controls whether [use idle connection reaper] is used
```
[the remark this answers is not in the document]
Comment (skytin1004) [reply docs/aws.md]: I fixed all the broken parts.

**Review by nastra — APPROVED** [review]
one minor nit but LGTM otherwise
